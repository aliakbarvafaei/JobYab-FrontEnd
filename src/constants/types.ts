export interface GeneralType {
  id: number;
  title: string;
}
type User = {
  activity_field: string;
  company_english_name: string;
  company_persian_name: string;
  company_phone_number: string;
  id: number;
  introduction: string;
  level: string;
  logo: string | null;
  number_of_personnel: string;
  type: string;
  user: number;
  website: string | null;
};
export type PostType = {
  id: number;
  city: GeneralType;
  cooperation_type: string;
  created_date: string;
  degree: string;
  description: string;
  experience: string;
  job_type: GeneralType;
  salary: string;
  sarbazi: string;
  sex: string;
  skills: GeneralType[];
  state: GeneralType;
  title: string;
  user: User;
  is_bookmark?: boolean;
};

type UserSpecificData = {
  date_joined: string;
  email: string;
  email_active_code: string;
  first_name: string;
  groups: string[];
  id: number;
  is_active: boolean;
  is_authorized: boolean;
  is_employer: boolean;
  is_staff: boolean;
  is_superuser: boolean;
  last_login: string;
  last_name: string;
  password: string;
  user_permissions: string[];
  username: string;
};
export type UserType = {
  address: string;
  phone_number: string;
  national_code: string;
  full_name: string;
  data: UserSpecificData;
};

export type StateType = {
  id: number;
  title: string;
  order: number;
};
