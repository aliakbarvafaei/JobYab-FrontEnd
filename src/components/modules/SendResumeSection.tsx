import { Button, InputBase, Typography } from "@mui/material";

const SendResumeSection = () => {
  return (
    <div
      style={{
        background: "#F2F3F4",
        width: "20%",
        marginTop: 40,
        borderRadius: 8,
        flexShrink: 0,
        paddingInline: 20,
        maxHeight: 370,
        position: "relative",
        boxShadow: "0 0 6px rgba(0, 0, 0, 0.2)",
      }}
    >
      <InputBase
        value="سید عرفان نوربخش "
        fullWidth
        style={{
          border: "2px solid gray",
          borderRadius: 8,
          textAlign: "center",
          marginTop: 25,
        }}
        disabled
        classes={{ disabled: "text-center" }}
      />
      <InputBase
        value="ErfanNourbakhsh@gmail.com"
        fullWidth
        style={{
          border: "2px solid gray",
          borderRadius: 8,
          textAlign: "center",
          marginTop: 25,
        }}
        disabled
        classes={{ disabled: "text-center" }}
      />
      <InputBase
        value="09134821558"
        fullWidth
        style={{
          border: "2px solid gray",
          borderRadius: 8,
          textAlign: "center",
          marginTop: 25,
        }}
        disabled
        classes={{ disabled: "text-center" }}
      />
      <div
        style={{
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center",
          marginTop: 25,
        }}
      >
        <Typography style={{ fontSize: 14, color: "gray" }}>
          آپلود رزومه:
        </Typography>
        <Button variant="contained" component="label">
          آپلود
          <input hidden accept="image/*" multiple type="file" />
        </Button>
      </div>
      <Button fullWidth variant="outlined" style={{ marginTop: 25 }}>
        ارسال رزومه
      </Button>
      <Typography
        style={{
          fontSize: 12,
          textAlign: "center",
          position: "absolute",
          bottom: 10,
          right: "17%",
        }}
      >
        فرصت ارسال رزومه:{" "}
        <span>
          تا
          <span style={{ color: "blue" }} children={` ${28} `} />
          روز دیگر
        </span>
      </Typography>
    </div>
  );
};
export default SendResumeSection;
