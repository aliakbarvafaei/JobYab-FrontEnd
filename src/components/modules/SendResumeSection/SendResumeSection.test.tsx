import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SendResumeSection from ".";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { UserType } from "../../../constants/types";

const mockStore = configureStore([]);

describe("SendResumeSection Component", () => {
  let store: any;
  let history: any;

  beforeEach(() => {
    store = mockStore({
      userAuth: {
        role: "admin",
        token: "some-token",
      },
    });
    history = createMemoryHistory();
  });
  test("submit button should be disabled when no file is uploaded", () => {
    render(
      <Provider store={store}>
        <Router history={history}>
          <SendResumeSection />
        </Router>
      </Provider>
    );
    const submitButton = screen.getByRole("button", { name: /ارسال رزومه/i });
    expect(submitButton).toBeInTheDocument();
    userEvent.click(submitButton);
    // Check that clicking the disabled button doesn't trigger any actions
  });
  test("input fields should display respective data correctly", () => {
    const mockData: UserType = {
      address: "تست",
      national_code: "1245764848",
      full_name: "John Doe",
      phone_number: "1234567890",
      data: {
        date_joined: "",
        email: "",
        email_active_code: "",
        first_name: "",
        groups: [],
        id: 1,
        is_active: false,
        is_authorized: false,
        is_employer: false,
        is_staff: false,
        is_superuser: false,
        last_login: "",
        last_name: "",
        password: "",
        user_permissions: [],
        username: "",
      },
    };
    render(
      <Provider store={store}>
        <Router history={history}>
          <SendResumeSection data={mockData} />
        </Router>
      </Provider>
    );
    const fullNameInput = screen.getByDisplayValue(/john doe/i);
    const phoneNumberInput = screen.getByDisplayValue("1234567890");
    expect(fullNameInput).toBeInTheDocument();
    expect(phoneNumberInput).toBeInTheDocument();
  });
});
