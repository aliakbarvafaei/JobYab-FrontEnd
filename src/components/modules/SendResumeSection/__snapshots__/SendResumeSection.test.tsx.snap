// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendResumeSection Component renders correctly in SendResumeSection Page 1`] = `
<DocumentFragment>
  <div
    class="lgmin:w-3/12 md:w-11/12 "
    style="background: rgb(242, 243, 244); margin-top: 40px; border-radius: 8px; flex-shrink: 0; max-height: 280px; position: relative; box-shadow: 0 0 6px rgba(0, 0, 0, 0.2);"
  >
    <div
      style="position: absolute; background: rgb(217, 221, 220); opacity: 0.7; width: 100%; height: 100%; border-radius: 8px; z-index: 9999;"
    />
    <div
      style="position: absolute; display: flex; justify-content: center; align-items: center; width: 100%; height: 100%; z-index: 99999; text-align: center;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="p-[20px]"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled text-center MuiInputBase-fullWidth css-luel6v-MuiInputBase-root"
        style="border: 2px solid gray; border-radius: 8px; text-align: center; margin-top: 25px;"
      >
        <input
          class="MuiInputBase-input Mui-disabled text-center css-yz9k0d-MuiInputBase-input"
          disabled=""
          type="text"
          value=""
        />
      </div>
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled text-center MuiInputBase-fullWidth css-luel6v-MuiInputBase-root"
        style="border: 2px solid gray; border-radius: 8px; text-align: center; margin-top: 25px;"
      >
        <input
          class="MuiInputBase-input Mui-disabled text-center css-yz9k0d-MuiInputBase-input"
          disabled=""
          type="text"
          value=""
        />
      </div>
      <div
        style="display: flex; justify-content: space-between; align-items: center; margin-top: 25px;"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          style="font-size: 14px; color: gray;"
        >
          آپلود رزومه:
        </p>
        <label
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          role="button"
          tabindex="0"
        >
          آپلود
          <input
            accept="application/pdf, application/vnd.ms-excel"
            hidden=""
            multiple=""
            type="file"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </label>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth css-xsqe0f-MuiButtonBase-root-MuiButton-root"
        style="margin-top: 25px; margin-bottom: 10px;"
        tabindex="0"
        type="button"
      >
        ارسال رزومه
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
